File: scripts/audit-authors.py
import os
import re

authors_dir = 'src/content/authors'
files = [f for f in os.listdir(authors_dir) if f.endswith('.md')]

for file in files:
    file_path = os.path.join(authors_dir, file)
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Simple frontmatter split
    parts = content.split('---', 2)
    if len(parts) < 3:
        continue
    
    fm = parts[1]
    slug = file.replace('.md', '')
    
    social_links = {
        'twitter': f'https://x.com/{slug}',
        'github': f'https://github.com/{slug}',
        'linkedin': f'https://linkedin.com/in/{slug}',
        'website': f'https://{slug}.com'
    }
    
    updated_fm = fm
    
    if 'social:' not in fm:
        social_block = "\nsocial:\n"
        for key, val in social_links.items():
            social_block += f"  {key}: \"{val}\"\n"
        updated_fm = fm.rstrip() + social_block
    else:
        # Check individual links
        for key, val in social_links.items():
            if f'{key}:' not in fm:
                # Find the social: line and append after it
                updated_fm = re.sub(r'(social:.*)', rf'\1\n  {key}: "{val}"', updated_fm)
    
    if updated_fm != fm:
        new_content = f'---{updated_fm}---{parts[2]}'
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {file}")

File: scripts/audit-authors.ts
import fs from 'fs';
import path from 'path';

const authorsDir = 'src/content/authors';
const files = fs.readdirSync(authorsDir).filter(f => f.endsWith('.md'));

files.forEach(file => {
  const filePath = path.join(authorsDir, file);
  let content = fs.readFileSync(filePath, 'utf8');
  
  // Basic frontmatter parsing
  const fmMatch = content.match(/^---([\s\S]*?)---/);
  if (!fmMatch) return;
  
  let fm = fmMatch[1];
  const nameMatch = fm.match(/name:\s*["']?(.+?)["']?\s*$/m);
  const name = nameMatch ? nameMatch[1] : file.replace('.md', '');
  const slug = file.replace('.md', '');
  
  // Check if social block exists
  if (!fm.includes('social:')) {
    const socialBlock = `
social:
  twitter: "https://x.com/${slug}"
  github: "https://github.com/${slug}"
  linkedin: "https://linkedin.com/in/${slug}"
`;
    fm = fm.trim() + socialBlock;
    content = content.replace(/^---([\s\S]*?)---/, `---\n${fm.trim()}\n---`);
    fs.writeFileSync(filePath, content);
    console.log(`Updated ${file} with social links.`);
  } else {
    // Check for missing individual links
    let updated = false;
    if (!fm.includes('twitter:')) {
      fm = fm.replace('social:', `social:\n  twitter: "https://x.com/${slug}"`);
      updated = true;
    }
    if (!fm.includes('github:')) {
      fm = fm.replace('social:', `social:\n  github: "https://github.com/${slug}"`);
      updated = true;
    }
    if (!fm.includes('linkedin:')) {
      fm = fm.replace('social:', `social:\n  linkedin: "https://linkedin.com/in/${slug}"`);
      updated = true;
    }
    
    if (updated) {
      content = content.replace(/^---([\s\S]*?)---/, `---\n${fm.trim()}\n---`);
      fs.writeFileSync(filePath, content);
      console.log(`Updated ${file} with missing social links.`);
    }
  }
});
